src/charm.py

```diff
@@ -109,6 +117,8 @@ def __init__(self, *args):
             chain=None,
         )
 
+        self.secret_store.migrate_private_key()
```

Comment:
**Question**: Is it safe to read/write `secret_store` backends in the constructor?

Not sure if:
- On install, can we already read/write secrets?
- Before pebble-ready, can we read/write stored state?
- (Nitpick) can we touch stored state/secrets on stop/remove?